File: src/falcon_core/autotuner_interfaces/names/gname.py
"""Group naming convention for indexingconfig files."""

from .name_base import NameBase


class Gname(NameBase):
    """Names of the form "group#" where # is a int."""

    _index_string = "group"

    def __init__(self, value: str | int):
        super().__init__(value)

    @property
    def gname(self) -> str:
        """Return the gname."""
        return self.name

File: src/autotuner_interfaces/names/Gname.cpp
#include "falcon_core/autotuner_interfaces/names/Gname.hpp"

namespace falcon_core::autotuner_interfaces::names {
Gname::Gname() : NameBase() {}
Gname::Gname(const std::string& name) : NameBase(name) {}
Gname::Gname(const int& num) : NameBase(num) {}
std::string Gname::gname() const { return name(); }
}  // namespace falcon_core::autotuner_interfaces::names
CEREAL_REGISTER_TYPE(falcon_core::autotuner_interfaces::names::Gname)
CEREAL_REGISTER_POLYMORPHIC_RELATION(
    falcon_core::autotuner_interfaces::names::NameBase,
    falcon_core::autotuner_interfaces::names::Gname)
